### ISSUE TITLE
panic: runtime error: index out of range

### ISSUE THREAD

**Issue by masterjg** [hashicorp/terraform#21013]
### Terraform Version

```
λ terraform -v
Terraform v0.12.0-beta1
+ provider.aws v1.60.0-dev20190216h00-dev
+ provider.null v2.1.0-dev20190216h00-dev
+ provider.random v3.0.0-dev20190216h01-dev
+ provider.template v2.1.0-dev20190216h00-dev
+ provider.tls v2.0.0-dev20190216h00-dev
```

### Terraform Configuration Files

```hcl
...
```

### Crash Output

https://gist.github.com/masterjg/b031da5944c117074d3a253729405243

### Steps to Reproduce

1. `terraform init`
2. `terraform apply`

**Comment by masterjg** [comment]
Destroy works however. Only on apply and plan if is reproducible.

**Comment by apparentlymart** [comment]
Hi @masterjg! Sorry for this crash, and thanks for reporting it.

It looks like you've hit the same issue as described in #20750 here, which was subsequently fixed in #20765 but not yet included in a follow-up release. The beta2 release is coming soon, which will include this fix.

If you'd like to try it in the mean time, you could try with the latest [dev snapshot](http://terraform-0.12.0-dev-snapshots.s3-website-us-west-2.amazonaws.com/terraform/), which was built yesterday and so should include this fix. The behavior of the "dev20190415H16" snapshot, along with some further fixes we're making beforehand, is what will be labeled as beta2.

Since the fix for this is already in master, I'm going to close it out. Thanks again for reporting this!
